MODULE.bazel

```diff
@@ -0,0 +1,23 @@
+module(
+    name = "io_bazel_rules_go",
```

Comment:
Can we use this opportunity to go to just `rules_go`? I think that's allowed for rules that live in the `bazelbuild` org.

Reply:
That is something else we can only do via a patch for now: Since there is no `repo_name` attribute for the main module, this has to remain identical to the `name` attribute in our `workspace` declaration. In the BCR, we will patch it to read `rules_go`.

Reply:
Is that a limitation of BCR and therefore something we should mostly ignore for now, or is this something where I should do some early legwork to get things sorted out?

Reply:
It's not a limitation of the BCR, rather it was a deliberate choice made by Xudong and Yun: Every Bazel module *except* the root module can specify a repo name that's different from the module name. If we were to change the name to `rules_go` here, the build would fail as the remaining references to `@io_bazel_rules_go` could not be resolved. In the BCR, we patch these to read `@rules_go` and thus can also change the module name to `rules_go`. I don't think we can do anything about this until the migration to `rules_go` has been completed for both `WORKSPACE` and `MODULE.bazel` consumers.